motionplan/motionPlanner.go

```diff
@@ -400,11 +400,17 @@ func (mp *planner) getSolutions(ctx context.Context, seed referenceframe.FrameSy
 	ctxWithCancel, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	solutionGen := make(chan *ik.Solution, mp.planOpts.NumThreads*2)
+	solutionGen := make(chan *ik.Solution, mp.planOpts.NumThreads*20)
 	ikErr := make(chan error, 1)
 	var activeSolvers sync.WaitGroup
 	defer activeSolvers.Wait()
 	activeSolvers.Add(1)
+	defer func() {
+		// In the case that we have an error, we need to explicitly drain the channel before we return
```

Comment:
This will execute even if we exit this function successfully and that seems like something we do not want

Reply:
If we exit the function successfully, `len(solutionGen)` must necessarily already be 0.